Use of deleted RationalConstant default constructor in Units constructor

Hit this compile error on a newly-updated Linux box with an extra persnickety C++ compiler:

```
/home/jeff/projects/sandia/scream/components/scream/src/share/util/units.hpp:55:31: error: use of deleted function ‘scream::util::RationalConstant::RationalConstant()’
   55 |    , m_exp_format (Format::Rat)
      |                               ^
In file included from /home/jeff/projects/sandia/scream/components/scream/src/share/util/units.hpp:4,
                 from /home/jeff/projects/sandia/scream/components/scream/src/share/field/field_identifier.hpp:6,
                 from /home/jeff/projects/sandia/scream/components/scream/src/share/field/field_header.hpp:4,
                 from /home/jeff/projects/sandia/scream/components/scream/src/share/field/field.hpp:4,
                 from /home/jeff/projects/sandia/scream/components/scream/src/share/util/scream_test_utils.hpp:6,
                 from /home/jeff/projects/sandia/scream/components/scream/src/share/util/scream_catch_main.cpp:7:
/home/jeff/projects/sandia/scream/components/scream/src/share/util/rational_constant.hpp:48:3: note: declared here
   48 |   RationalConstant () = delete;
```

Looks like we need to be more specific about how to initialize the `m_exp_format` field in the Unit class for that constructor. Anyone have any thoughts? I'm not familiar enough with how this stuff is used to have an opinion yet.

--- a/components/scream/src/share/util/rational_constant.hpp
+++ b/components/scream/src/share/util/rational_constant.hpp
@@ -70,7 +70,6 @@ public:
     // Nothing to do here
   }
   constexpr RationalConstant (const RationalConstant&) = default;
-  RationalConstant& operator= (const RationalConstant&) = default;
 
   constexpr RationalConstant operator- () const {
     return RationalConstant(-m_num,m_den,m_fmt);

--- a/components/scream/src/share/util/scaling_factor.hpp
+++ b/components/scream/src/share/util/scaling_factor.hpp
@@ -31,7 +31,6 @@ struct ScalingFactor {
   }
 
   constexpr ScalingFactor (const ScalingFactor&) = default;
-  ScalingFactor& operator= (const ScalingFactor&) = default;
 
   static constexpr ScalingFactor one () { return ScalingFactor(1); }
   static constexpr ScalingFactor zero () { return ScalingFactor(0); }

--- a/components/scream/src/share/util/units.hpp
+++ b/components/scream/src/share/util/units.hpp
@@ -76,7 +76,6 @@ public:
   }
 
   Units (const Units&) = default;
-  Units& operator= (const Units&) = default;
 
   static Units nondimensional () {
     return Units(RationalConstant::one());

--- a/components/scream/src/share/util/units.hpp
+++ b/components/scream/src/share/util/units.hpp
@@ -51,7 +51,9 @@ public:
   // Construct a non-dimensional quantity
   Units (const ScalingFactor& scaling)
    : m_scaling {scaling}
-   , m_units{0,0,0,0,0,0,0}
+   , m_units{RationalConstant(0), RationalConstant(0), RationalConstant(0),
+             RationalConstant(0), RationalConstant(0), RationalConstant(0),
+             RationalConstant(0)}
    , m_exp_format (Format::Rat)
   {
     m_string = to_string(*this);

